pkg/internal/diskutil.go

```diff
@@ -0,0 +1,224 @@
+	}
+	return v, err
+}
+
+// GetReadOnly as bool
+func (b BlockDevice) GetReadOnly() (bool, error) {
+	v, err := parseBitBool(b.ReadOnly)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to parse readOnly property %q as bool", b.ReadOnly)
+	}
+	return v, err
+}
+
+// GetRemovable as bool
+func (b BlockDevice) GetRemovable() (bool, error) {
+	v, err := parseBitBool(b.Removable)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to parse removable property %q as bool", b.Removable)
+	}
+	return v, err
+}
+
+func parseBitBool(s string) (bool, error) {
+	if s == "0" {
+		return false, nil
+	} else if s == "1" {
+		return true, nil
+	}
+	return false, fmt.Errorf("lsblk bool value not 0 or 1: %q", s)
+}
+
+// GetSize as int64
+func (b BlockDevice) GetSize() (int64, error) {
+	v, err := strconv.ParseInt(b.Size, 10, 64)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to parse size property %q as int64", b.Size)
+	}
+	return v, err
+}
+
+// HasChildren check on BlockDevice
+func (b BlockDevice) HasChildren() (bool, error) {
+	sysDevDir := filepath.Join("/sys/block/", b.Name, "/*")
+	paths, err := filepath.Glob(sysDevDir)
+	if err != nil {
+		return false, errors.Wrapf(err, "could not glob path: %q to verify partiions", sysDevDir)
+	}
+	for _, path := range paths {
+		name := filepath.Base(path)
+		if strings.HasPrefix(name, b.Name) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// GetPathByID check on BlockDevice
+func (b BlockDevice) GetPathByID() (string, error) {
+
+	// return if previously populated value is valid
+	if len(b.pathByID) > 0 && strings.HasPrefix(b.pathByID, DiskByIDDir) {
+		evalsCorrectly, err := PathEvalsToDiskLabel(b.pathByID, b.Name)
+		if err == nil && evalsCorrectly {
+			return b.pathByID, nil
+		}
+	}
+	b.pathByID = ""
+	diskByIDDir := filepath.Join(DiskByIDDir, "/*")
+	paths, err := filepath.Glob(diskByIDDir)
+	if err != nil {
+		return "", fmt.Errorf("could not list files in %q: %w", DiskByIDDir, err)
+	}
+	for _, path := range paths {
+		isMatch, err := PathEvalsToDiskLabel(path, b.Name)
+		if err != nil {
+			return "", err
+		}
+		if isMatch {
+			b.pathByID = path
+			return path, nil
+		}
+	}
+	// return path by label and error
+	b.pathByID = fmt.Sprintf("/dev/%s", b.Name)
+	return b.pathByID, IDPathNotFoundError{DeviceName: b.Name}
+}
+
+// PathEvalsToDiskLabel checks if the path is a symplink to a file devName
+func PathEvalsToDiskLabel(path, devName string) (bool, error) {
+	devPath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return false, errors.Wrapf(err, "could not eval symLink %q", path)
+	}
+	if filepath.Base(devPath) == devName {
+		return true, nil
+	}
+	return false, nil
+}
+
+// ListBlockDevices using the lsblk command
+func ListBlockDevices() ([]BlockDevice, []string, error) {
+	var output bytes.Buffer
+	var blockDevices []BlockDevice
+
+	columns := "NAME,ROTA,TYPE,SIZE,MODEL,VENDOR,RO,RM,STATE,FSTYPE"
+	args := []string{"--pairs", "-b", "-o", columns}
```

Comment:
There is an option for using `--json`.  Could that be more useful here ?
`lsblk --json --output  "SIZE,ROTA,RO,TYPE,PKNAME,NAME,VENDOR,ROTA,FSTYPE,MODEL"`

Reply:
won't be available in lsblk until we start using rhel8 based images

Reply:
I started with that, and the rewrote it :grimacing: